src/serverenvironment.cpp

```diff
@@ -495,71 +515,77 @@ void ServerEnvironment::kickAllPlayers(AccessDeniedCode reason,
 
 void ServerEnvironment::saveLoadedPlayers()
 {
-	std::string players_path = m_path_world + DIR_DELIM "players";
+	std::string players_path = m_path_world + DIR_DELIM + "players";
 	fs::CreateDir(players_path);
 
 	for (std::vector<RemotePlayer *>::iterator it = m_players.begin();
 		it != m_players.end();
 		++it) {
 		if ((*it)->checkModified() ||
 			((*it)->getPlayerSAO() && (*it)->getPlayerSAO()->extendedAttributesModified())) {
-			(*it)->save(players_path, m_server);
+			try {
+				if (!m_player_database->savePlayer(*it)) {
+					errorstream << "Unable to save player " << (*it)->getName()
+							<< ", SAO was not found." << std::endl;
```

Comment:
it's not so simple, DatabaseExceptions are triggered at request time, false is only triggered when SAO is not available, but i can return and we can ignore error, SAO unavailable on save is not possible with the refactor, i will do this